### PR TITLE
Add Hessen DTM and DSM

### PR DESCRIPTION
**Pull request by anikaweinmann**
Add Hessen DTM and DSM tindex. The GHA workflow does not run successfully (it get an timeout after 6 h). Locally the scripts needs only some minutes to generate the tindices.

### CODE REVIEW

File: DSM/HE/HE_DSM_tindex.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+        gem_vars.append(f"{gem} am Rhein")
+    if gem == "Heppenheim":
+        gem_vars.append(f"{gem} (Bergstraße)")
+    return gem_vars
+
+
+"""MAIN PART"""
+
+os.chdir("/src/tile-indices/DSM/HE/")
+# os.chdir("DSM/HE/")
```
Comment (linakrisztian) [inline DSM/HE/HE_DSM_tindex.py:170]: The commented path, would be as within this repo. If adjustable, maybe also move to top

File: DSM/HE/HE_DSM_tindex.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+create_gz = stream.read()
+print(f"<{gpkg_name}.gz> created")
+
+# cleanup
+if os.path.isdir("tmp"):
+    shutil.rmtree("tmp")
+if os.path.isfile(geojson_name):
+    os.remove(geojson_name)
+if os.path.isfile(gpkg_name):
+    os.remove(gpkg_name)
```
Comment (linakrisztian) [inline DSM/HE/HE_DSM_tindex.py:336]: maybe good to have the cleanup not only in the end, but also if script fails (e.g. with `atexit`).

File: DTM/HE/HE_DTM_tindex.py [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+############################################################################
+
+# Usage inside grass location:
+#   grass -c epsg:25832 /grassdb/HE_ALKIS --exec python3 DTM/HE/HE_DTM_tindex.py
```
Comment (linakrisztian) [inline DTM/HE/HE_DTM_tindex.py:30]:
```suggestion
#   grass -c epsg:25832 /grassdb/HE_DTM --exec python3 DTM/HE/HE_DTM_tindex.py
```

File: DTM/HE/HE_DTM_tindex.py [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+    tries = 0
+    while tries <= MAX_RETRIES and test_url == 400:
+        try:
+            test_url = urlopen(url_today, timeout=600).getcode()
+        except HTTPError as e:
+            # HTTPError means 404
+            tries = 100
+            pass
+        except URLError as e:
+            if tries == 14:
```
Comment (linakrisztian) [inline DTM/HE/HE_DTM_tindex.py:93]: see above

File: DTM/HE/HE_DTM_tindex.py [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+        gem_vars.append(f"{gem} am Rhein")
+    if gem == "Heppenheim":
+        gem_vars.append(f"{gem} (Bergstraße)")
+    return gem_vars
+
+
+"""MAIN PART"""
+
+os.chdir("/src/tile-indices/DTM/HE/")
+# os.chdir("DTM/HE/")
```
Comment (linakrisztian) [inline DTM/HE/HE_DTM_tindex.py:170]: see above

File: DTM/HE/HE_DTM_tindex.py [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+print(tindex_verification)
+
+# package
+if os.path.isfile(f"{gpkg_name}.gz"):
+    os.remove(f"{gpkg_name}.gz")
+stream = os.popen(f"gzip {gpkg_name}")
+create_gz = stream.read()
+print(f"<{gpkg_name}.gz> created")
+
+# cleanup
```
Comment (linakrisztian) [inline DTM/HE/HE_DTM_tindex.py:330]: see above

**Review by linakrisztian — APPROVED** [review]
Thanks a lot! Looks good to me, just some comments.
Maybe in future we could exclude some part as functions to reduce duplicates, but should be fine for now.

File: .github/workflows/tindex_dsm_dtm_he.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+name: Update DSM and DTM HE Tile Index
+# funktioniert bisher nur lokal und nicht in GHA (da gibt es nach 6 h einen timeout)
+
+on:
+  push:
+    branches: [ HE_DTM_DSM ]
```
[the remark this answers is not in the document]
Comment (anikaweinmann) [reply .github/workflows/tindex_dsm_dtm_he.yml:6]: I would remove it if it is fixed. Leaving it it should avoid that the GHA workflow runs on main

File: DSM/HE/HE_DSM_tindex.py [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+def check_url(url):
+    """Function to check if url is reachable"""
+    url_today = url.replace("DATE", date)
+    test_url = 400
+    tries = 0
+    while tries <= MAX_RETRIES and test_url == 400:
+        try:
+            test_url = urlopen(url_today, timeout=600).getcode()
+        except HTTPError as e:
+            # HTTPError means 404
```
[the remark this answers is not in the document]
Comment (anikaweinmann) [reply DSM/HE/HE_DSM_tindex.py:89]: Then you get the wrong urls. I think they are not interesting for us.

**Pull request merged**